src/spdx_tools/common/typing/dataclass_with_properties.py

```diff
@@ -26,11 +28,11 @@ def set_field(self, value: field_type):
     def set_field_with_better_error_message(self, value: field_type):
         try:
             set_field(self, value)
-        except TypeError as err:
-            error_message: str = f"SetterError {self.__class__.__name__}: {err.args[0]}"
+        except TypeCheckError as err:
+            error_message: str = f"SetterError {self.__class__.__name__}: {err}"
             # As setters are created dynamically, their argument name is always "value". We replace it by the
             # actual name so the error message is more helpful.
-            raise TypeError(error_message.replace("value", field_name, 1) + f": {value}")
+            raise TypeError(error_message.replace("value", field_name, 1))
```

Comment:
I'm not sure if it is bad practice to simply change the error type while reraising.
But all tests currently expect a `TypeError` and not a `TypeCheckError`, so your approach here certainly is a less breaking change.